multimodal/tests/unittests/others/test_metrics.py

```diff
@@ -105,3 +112,48 @@ def test_symmetric_hit_rate():
             hit_rate_impl = compute_hit_rate(features_a, features_b, logit_scale=1.0, top_ks=top_ks)
             hit_rate_ref = ref_symmetric_hit_rate(features_a, features_b, logit_scale=1.0, top_ks=top_ks)
             assert pytest.approx(hit_rate_impl.item()) == hit_rate_ref.item()
+
+
+def test_custom_metric():
+    dataset = HatefulMeMesDataset()
+    metric_name = dataset.metric
+    metric_scorer = ag_metrics.get_metric(metric_name)
+    predictor_by_name = MultiModalPredictor(
+        label=dataset.label_columns[0],
+        problem_type=dataset.problem_type,
+        eval_metric=metric_name,
+    )
+    predictor_by_scorer = MultiModalPredictor(
+        label=dataset.label_columns[0],
+        problem_type=dataset.problem_type,
+        eval_metric=metric_scorer,
+    )
+    with tempfile.TemporaryDirectory() as save_path:
+        if os.path.isdir(save_path):
+            shutil.rmtree(save_path)
+        predictor_by_name.fit(
+            train_data=dataset.train_df,
+            time_limit=10,
+            save_path=save_path,
+        )
+    with tempfile.TemporaryDirectory() as save_path:
+        if os.path.isdir(save_path):
+            shutil.rmtree(save_path)
+        predictor_by_scorer.fit(
+            train_data=dataset.train_df,
+            time_limit=10,
+            save_path=save_path,
+        )
+    scores_by_name = predictor_by_name.evaluate(
+        data=dataset.test_df,
+        metrics=None,
+    )
+    scores_by_scorer_eval = predictor_by_name.evaluate(
+        data=dataset.test_df,
+        metrics=[metric_scorer],
+    )
+    scores_by_scorer_init = predictor_by_scorer.evaluate(
+        data=dataset.test_df,
+        metrics=None,
+    )
+    assert scores_by_name[metric_name] == scores_by_scorer_eval[metric_name] == scores_by_scorer_init[metric_name]
```

Comment:
Is it safe to asser `scores_by_name[metric_name]==scores_by_scorer_init[metric_name]`? This requires the models in `predictor_by_name` and `predictor_by_score` have exactly the same state. Although use the same time limit for them, can we guarantee the model is the same?